aten/src/ATen/native/cuda/ReduceLogicKernel.cu

```diff
@@ -24,4 +26,24 @@ void or_kernel_cuda(TensorIterator& iter) {
 REGISTER_DISPATCH(and_stub, &and_kernel_cuda);
 REGISTER_DISPATCH(or_stub, &or_kernel_cuda);
 
+bool cuda_equal(const Tensor& self, const Tensor &src) {
+  if (!at::namedinference::are_names_equal(
+        self.unsafeGetTensorImpl(), src.unsafeGetTensorImpl())) {
+    return false;
+  }
+  at::NoNamesGuard guard;
+  TORCH_CHECK(self.device() == src.device(), "Cannot compare two tensors on "
+      "different devices. Got: ", self.device(), " and ", src.device());
+  if (self.sizes() != src.sizes()) {
+    return false;
+  }
+  if (self.numel() == 0) {
+    return true;
+  }
+  return THCudaBoolTensor_minall(
+      globalContext().getTHCState(),
+      at::native::eq(self, src).unsafeGetTensorImpl()
```

Comment:
It is safer if you call: `at::native::eq(self, src).min()`

Reply:
That leads to serious performance regression.

Reply:
Sorry for going circles but can you please check with `at::native::eq(self, src).all()` https://pytorch.org/docs/master/tensors.html#torch.BoolTensor.all

Reply:
I'm shocked. Not only is there no regression, it seems to be significantly faster now. (Updated the benchmarks in PR message)